### PR TITLE
Added mDNS support for AP interface

### PR DESCRIPTION
**Pull request by Rocketct**
Added mDNS support for AP interface to allow the hostname management when AP configuration is enabled

cc:/ @facchinm @sandeepmistry test and take a look on the sdkconfig there is somethings of new
i had just make some tests and seems that works fine, i tried also to connect two device in the same time and all goes fine

this fix https://github.com/arduino-libraries/WiFiNINA/issues/92 for 1010, Iot 33 and works also on uno WiFi

### CODE REVIEW

File: sdkconfig [hunk 1]
```diff
@@ -206,6 +206,12 @@ CONFIG_EFUSE_MAX_BLK_LEN=192
 # ESP32-specific
 #
 CONFIG_IDF_TARGET_ESP32=y
+CONFIG_ESP32_REV_MIN_0=y
```
Comment (sandeepmistry) [inline sdkconfig]: Which IDF are you using? If it's an IDF update, it should be an independent commit.

Comment (sandeepmistry) [reply sdkconfig]: Are these all the default values?

File: arduino/libraries/WiFi/src/WiFi.cpp [hunk 2]
```diff
@@ -697,6 +703,15 @@ void WiFiClass::handleSystemEvent(system_event_t* event)
     case SYSTEM_EVENT_AP_START: {
       struct netif* apNetif;
 
+      mdns_init();
```
Comment (sandeepmistry) [inline arduino/libraries/WiFi/src/WiFi.cpp]: Are you sure this `mdns_init` is needed? I believe without it macOS was fine in STA mode, should we also port to STA mode?

File: arduino/libraries/WiFi/src/WiFi.cpp [hunk 3]
```diff
@@ -705,6 +720,8 @@ void WiFiClass::handleSystemEvent(system_event_t* event)
         }
       }
 
+      mdns_hostname_set(_hostname);
```
Comment (sandeepmistry) [inline arduino/libraries/WiFi/src/WiFi.cpp]: What's the difference between `mdns_hostname_set` and `tcpip_adapter_set_hostname`?

File: arduino/libraries/WiFi/src/WiFi.cpp [hunk 4]
```diff
@@ -32,6 +32,12 @@
 #include <lwip/inet_chksum.h>
 
 #include "WiFi.h"
+#include "mdns.h"
+
+// defines required from mdns service
+#define LWIP_MDNS_RESPONDER             1
+#define LWIP_IGM                        1
+#define LWIP_IPV4                       1
```
Comment (sandeepmistry) [inline arduino/libraries/WiFi/src/WiFi.cpp]: It seems really strange we need these defines to be manually added ... are you sure we need them?

File: arduino/libraries/WiFi/src/WiFi.cpp (hunk 2 above)
Comment (Rocketct) [reply arduino/libraries/WiFi/src/WiFi.cpp]: As described by the example seems yes, its recommend by the example, i had test adding in the properly interface like how we do in STA but not works i had found an issue on the epressif repo.
This is why i suppose to test the mdns and from the result was that works for the access point.
We can try to move on other direction implement in STA mode and removing the interface management if a plus.

Comment (Rocketct) [reply arduino/libraries/WiFi/src/WiFi.cpp]: here the example: https://github.com/espressif/esp-idf/blob/master/examples/protocols/mdns/main/mdns_example_main.c

File: sdkconfig (hunk 1 above)
Comment (Rocketct) [reply sdkconfig]: no removed and enabled by adding the entries on the kconfig on the idf folder (modifying the kconfig) say me if wrong procedure

File: arduino/libraries/WiFi/src/WiFi.cpp (hunk 2 above)
Comment (Rocketct) [reply arduino/libraries/WiFi/src/WiFi.cpp]: As i understand when the AP interface is used we should enable a dns local instance because we don't have a reference DNS Server like the STA (i'm testing it).
Now i'm try to correctly configure the local dns and ad the feature `AS_DYNAMIC` in order to be able to manage the lookup table and the new rules if this works we could leave the idea  of implement the mdns.

File: arduino/libraries/WiFi/src/WiFi.cpp (hunk 3 above)
Comment (Rocketct) [reply arduino/libraries/WiFi/src/WiFi.cpp]: mdns_hostname_set set the name on the mdns onject the other goes to set the hostname on the lwip tcpip interface

Comment (Rocketct) [reply arduino/libraries/WiFi/src/WiFi.cpp]: maybe the second is useless while mdns is enabled

File: arduino/libraries/WiFi/src/WiFi.cpp (hunk 4 above)
Comment (Rocketct) [reply arduino/libraries/WiFi/src/WiFi.cpp]: removed!! added on the kconfig

File: arduino/libraries/WiFi/src/WiFi.cpp (hunk 3 above)
Comment (Rocketct) [reply arduino/libraries/WiFi/src/WiFi.cpp]: in STA mode mdns not works

**Comment by Rocketct** [comment]
closed the issue because the actual behavior match with the one of the mkr1000
